internal/run/run.go

```diff
@@ -905,9 +899,39 @@ func resolveRemoteDocument(ctx context.Context, d workflow.Document, outPath str
 		token = os.Getenv(strings.ToUpper(envVar))
 	}
 
-	if err := download.DownloadFile(d.Location, outPath, header, token); err != nil {
+	res, err := download.Fetch(d.Location, header, token)
+	if err != nil {
 		return "", err
 	}
+	defer res.Body.Close()
+
+	ext := filepath.Ext(outPath)
+	if !slices.Contains([]string{".yaml", ".yml", ".json"}, ext) {
```

Comment:
If we're provided with an output path and that path has a recognisable extension then we don't need to apply any of our heuristics. We'll respect that e.g.

```yaml
sources:
    petstore:
        inputs:
            - location: https://api.example.com
        output: "./swagger.yaml"  # We're going to go with `.yaml`
```